File: programmers/lv0/코딩기초트레이닝/조건문/주사위게임2/Solution.cpp
#include <string>
#include <cmath>

using namespace std;

int solution(int a, int b, int c) {
    int sum = a + b + c;
    int squareSum = pow(a, 2) + pow(b, 2) + pow(c, 2);
    int cubeSum = pow(a, 3) + pow(b, 3) + pow(c, 3);
    if (a != b && b != c && c != a) {
        return sum;
    }
    else if (a == b && b == c) {
        return sum * squareSum * cubeSum;
    }
    else {
        return sum * squareSum;
    }
}

File: programmers/lv0/코딩기초트레이닝/조건문/주사위게임2/Solution.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int solution(int a, int b, int c)
    {
        var set = new HashSet<int>{ a, b, c };
        int sum = a + b + c;
        int squareSum = Convert.ToInt32(Math.Pow(a, 2) + Math.Pow(b, 2) + Math.Pow(c, 2));
        int cubeSum = Convert.ToInt32(Math.Pow(a, 3) + Math.Pow(b, 3) + Math.Pow(c, 3));
        switch (set.Count)
        {
            case 1:
                return sum * squareSum * cubeSum;
            case 2:
                return sum * squareSum;
            case 3:
                return sum;
            default:
                return 0;
        }
    }
}